pkg/controller/workflow/context.go

```diff
@@ -134,6 +134,15 @@ func (c *Context) HasReason(reason ConditionReason) bool {
 	return false
 }
 
+func (c *Context) HasConditionType(condition status.ConditionType) bool {
```

Comment:
nit: this looks redundant to https://github.com/mongodb/mongodb-atlas-kubernetes/blob/main/pkg/api/v1/status/condition.go#L128

Reply:
This was added before rebasing on main. Removed.